### ISSUE TITLE
Password not working

### ISSUE THREAD

**Issue by snake3ca** [joshkunz/ashuffle#15]
Hi great application for MPD works great except I have one issue.

ashuffle tells me it cant play the track after adding a new song.  Seems to be related to the password.

I use ashuffle with the command: env MPD_HOST="songs@192.168.0.109" MPD_PORT="6600" ./ashuffle

When the playing track finishes then it adds a new track then wont play it.  
I get the error: MPD error: you don't have permission for "play"

The password I am using is correct and works on all my mpdclients.

I am using the newest MPD build 0.21.3

**Comment by snake3ca** [comment]
Also I am using ashuffle 1.1

**Comment by joshkunz** [comment]
Hi, any chance you should share your MPD config? Feel free to remove anything personally identifying/change passwords whatever. It would just make it easier to try and reproduce myself.

**Comment by joshkunz** [comment]
Thanks for the config! I’ll see if I can reproduce this myself. I don’t use MPD’s password feature myself, so I’m not particuilarly familiar with it. Don’t worry about the formatting either, I edited your post to fix it.

**Comment by purrloftruth** [comment]
I'm experiencing the same issue on FreeBSD 11.2. Notably, mpc works just fine with exactly the same MPD_HOST env variable set.

```
# Files and directories #######################################################
#

music_directory		"/spaces/music"
playlist_directory		"/var/mpd/.mpd/playlists"
db_file			"/var/mpd/.mpd/database"
log_file			"/var/mpd/.mpd/log"
pid_file			"/var/mpd/.mpd/pid"
state_file			"/var/mpd/.mpd/state"
sticker_file			"/var/mpd/.mpd/sticker.sql"

#
###############################################################################


# General music daemon options ################################################
#

user				"mpd"
group				"mpd"
bind_to_address		"any"
port				"6600"
auto_update	"yes"

#
###############################################################################


# Symbolic link behavior ######################################################
#

follow_outside_symlinks	"yes"
follow_inside_symlinks		"yes"

#
###############################################################################


# Permissions #################################################################
#

password                        "<removed>@read,add,control,admin"
password                        "<removed>@read,add,control"
default_permissions             "read"

#
###############################################################################


# Input #######################################################################
#

input {
        plugin "curl"
}

#
###############################################################################

# Audio Output ################################################################
#

audio_output {
	type			"shout"
	encoder		"mp3"			# optional
	name			"<removed>"
	host			"localhost"
	port			"8000"
	mount		"/automation"
	password		"<removed>"
	bitrate		"192"
	format		"44100:16:2"
	protocol		"icecast2"		# optional
	always_on	"yes"
}

#
###############################################################################
```

...and some stuff about ReplayGain that I doubt is relevant...

**Comment by joshkunz** [comment]
I was able to reproduce this independently, the linked commit (ae20973) has more details. In short, `ashuffle` was previously only applying the password supplied by the user if the `status` command failed. Since both of these configurations enable `status` for the default user, `ashuffle` would not detect that a password was needed, and wouldn’t apply the given password.

This is fixed in `v1.0.2`. Now passwords supplied by the user are always sent to MPD. `ashuffle` also has better logic now for detecting if it has the appropriate permissions to run.

@snake3ca and @purrloftruth, thanks for filling/commenting on the issue. This should be fixed now, please re-open the issue if it isn’t working for you.
